swift-coverage/swift-coverage-example/xccov-to-sonarqube-generic.sh

```diff
@@ -16,7 +21,12 @@ function convert_file {
 function xccov_to_generic {
   echo '<coverage version="1">'
   for xccovarchive_file in "$@"; do
-    xcrun xccov view --file-list "$xccovarchive_file" | while read -r file_name; do
+    local file_list_cmd="xcrun xccov view"
+    if [[ $@ == *".xcresult"* ]]; then
```

Comment:
I think the if condition should be `$xccovarchive_file ==`  instead of `$@ ==`, because `$@` represent all the files and `$xccovarchive_file` one of the files.
I would prefer not having to build and eval the full xcrun command, but just a variable with additional xccov options like `--archive`.
@emanuelschmoczer what do you think about changing your PR to something like:
```bash
#!/usr/bin/env bash
set -euo pipefail

function convert_file {
  local xccovarchive_file="$1"
  local file_name="$2"
  local xccov_options="$3"
  echo "  <file path=\"$file_name\">"
  xcrun xccov view $xccov_options --file "$file_name" "$xccovarchive_file" | \
    sed -n '
    s/^ *\([0-9][0-9]*\): 0.*$/    <lineToCover lineNumber="\1" covered="false"\/>/p;
    s/^ *\([0-9][0-9]*\): [1-9].*$/    <lineToCover lineNumber="\1" covered="true"\/>/p
    '
  echo '  </file>'
}

function xccov_to_generic {
  echo '<coverage version="1">'
  for xccovarchive_file in "$@"; do
    local xccov_options=""
    if [[ $xccovarchive_file == *".xcresult"* ]]; then
      xccov_options="--archive"
    fi
    xcrun xccov view $xccov_options --file-list "$xccovarchive_file" | while read -r file_name; do
      convert_file "$xccovarchive_file" "$file_name" "$xccov_options"
    done
  done
  echo '</coverage>'
}

xccov_to_generic "$@"
```
Does it do the same?

Reply:
Thanks @alban-auzeill, having an xcov options variable is definitely the better solution. I can confirm that it works as expected and just updated the pr.

Reply:
@emanuelschmoczer thanks for your contribution :+1: